testing/fuchsia/run_tests.py

```diff
@@ -25,32 +41,89 @@
 from run_executable_test import ExecutableTestRunner
 from test_runner import TestRunner
 
-# TODO(https://github.com/flutter/flutter/issues/140179): Respect build
-# configurations.
-OUT_DIR = os.path.join(DIR_SRC_ROOT, 'out/fuchsia_debug_x64')
+if len(sys.argv) == 2:
+  VARIANT = sys.argv[1]
+  sys.argv.pop()
+elif len(sys.argv) == 1:
+  VARIANT = 'fuchsia_debug_x64'
+else:
+  assert False, 'Expect only one parameter as the compile output directory.'
+OUT_DIR = os.path.join(DIR_SRC_ROOT, 'out', VARIANT)
 
 
-# TODO(https://github.com/flutter/flutter/issues/140179): Execute all the tests
-# in
-# https://github.com/flutter/engine/blob/main/testing/fuchsia/test_suites.yaml
-# and avoid hardcoded paths.
-def _get_test_runner(runner_args: argparse.Namespace, *_) -> TestRunner:
-  return ExecutableTestRunner(
-      OUT_DIR, [],
-      'fuchsia-pkg://fuchsia.com/dart_runner_tests#meta/dart_runner_tests.cm',
-      runner_args.target_id, None, '/tmp/log',
-      [os.path.join(OUT_DIR, 'dart_runner_tests.far')], None
+class BundledTestRunner(TestRunner):
+
+  # private, use bundled_test_runner_of function instead.
+  def __init__(
+      self, target_id: str, package_deps: List[str], tests: List[str],
+      logs_dir: str
+  ):
+    super().__init__(OUT_DIR, [], None, target_id, package_deps)
+    self.tests = tests
+    self.logs_dir = logs_dir
+
+  def run_test(self) -> CompletedProcess:
+    returncode = 0
+    for test in self.tests:
+      # pylint: disable=protected-access
+      test_runner = ExecutableTestRunner(
+          OUT_DIR, [], test, self._target_id, None, self.logs_dir, [], None
+      )
+      test_runner._package_deps = self._package_deps
+      result = test_runner.run_test().returncode
+      logging.warning('Result of test %s is %s', test, result)
+      if result != 0:
+        returncode = result
+    return CompletedProcess(args='', returncode=returncode)
+
+
+def bundled_test_runner_of(target_id: str) -> BundledTestRunner:
+  log_dir = os.environ.get('FLUTTER_LOGS_DIR', '/tmp/log')
+  with open(os.path.join(os.path.dirname(__file__), 'test_suites.yaml'),
+            'r') as file:
+    tests = yaml.safe_load(file)
+  # Ignore tests needing multiple packages or with extra test arguments for
```

Comment:
Should this be a TODO? It seems like in it's current state this can run about half the tests.

Reply:
Ooops, yes, this should be a TODO.